src/main.rs

```diff
@@ -0,0 +1,107 @@
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{MerkleError, NodeIndex},
+    Felt, Word, ONE,
+    {hash::rpo::Rpo256, merkle::TieredSmt},
+};
+use std::time::Instant;
+
+fn main() {
+    construction(1000).unwrap();
+    construction(1_000_000).unwrap();
+
+    insertion(1000).unwrap();
+    insertion(1_000_000).unwrap();
+
+    proof_generation(1000).unwrap();
+    proof_generation(1_000_000).unwrap();
+}
+
+pub fn construction(size: u64) -> Result<(), MerkleError> {
```

Comment:
We can probably return the tree from here so that we don't have to construct it again in insert/prove benchmarks.